x-pack/test/security_solution_endpoint/services/endpoint_policy.ts

```diff
@@ -33,7 +33,7 @@ const INGEST_API_PACKAGE_POLICIES = `${INGEST_API_ROOT}/package_policies`;
 const INGEST_API_PACKAGE_POLICIES_DELETE = `${INGEST_API_PACKAGE_POLICIES}/delete`;
 const INGEST_API_EPM_PACKAGES = `${INGEST_API_ROOT}/epm/packages`;
 
-const SECURITY_PACKAGES_ROUTE = `${INGEST_API_EPM_PACKAGES}?category=security`;
+const SECURITY_PACKAGES_ROUTE = `${INGEST_API_EPM_PACKAGES}?category=security&experimental=true`;
```

Comment:
I was having the issue where 8.4.0-dev of the endpoint package was being installed, but this function was returning 8.3.0 as the latest version, adding experimental fixes this, but do we only ever want to install GA packages for these tests?

Reply:
cc @paul-tavares is installing experimental (prereleases) endpoint package a good idea for these tests?

_somewhat_ unlikely that a prerelease package is going to break tests since it's mostly mappings, but transform stuff is there too.

And the prereleases are going to be _more_ unstable now with automatic deploys.

on the face of it it doesn't seem great, but I'm having trouble thinking through the implications of what this affects

Reply:
@pzl ,
I think running against the latest package here is ok. Finding and addressing failures as soon as possible rather than to wait until a release package is available would be best. I'm actually a bit surprised - have we been testing against an older package all along?



@hop-dev ,
What does `experimental` do from a fleet API standpoint?
Our tests all run with a custom package registry running in docker (we get the config for the FTR tests from [here](https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/test/security_solution_endpoint_api_int/registry.ts#L40-L60), which uses the Docker image defined via the [fleet API integration config](https://github.com/elastic/kibana/blob/f5688a68a5ee27573c3ee36bc4652654c3e60729/x-pack/test/fleet_api_integration/config.ts#L16-L21). Just want to ensure that your change here will not cause the tests to become flaky, especially if somehow this flag is causing the API to reach out to the "real" registry (we have had problems in the past with CI test runs making external network calls to the real registry (especially network timeout or connection issues)).

Reply:
Hi @paul-tavares, adding `experimental` means EPR returns the latest version of a package it has whether it is GA or not, e.g it will return 8.4.0-dev instead of 8.3.0 of the endpoint package. It doesnt change where we go to get the packages.

As part of this PR I have moved to v2 of the EPR docker image which contains dev packages which it seems the previous v1 docker image we were using didn't. When the security tests start we install the latest version of the package [here](https://github.com/hop-dev/kibana/blob/0b162b68815e4880c7e784da9b89e63728b98e6e/x-pack/test/security_solution_endpoint/config.ts#L48) 

the previous issue was that we were installing 8.4.0-dev but then the EPR call was returning 8.3.0 as the latest version, adding experimental solved this.

If it is too risky to use the latest version I could look at a way of only specifying the latest GA release of the package on startup

Reply:
Just a note, the version of the package would only change if we update the docker image so I wouldnt expect this to add flakiness. Only the PR changing the image would fail

Reply:
^ got it. So this is still pulling a particular docker image to test from, and the query here is being sent to that point-in-time image. Sounds like there will be at least some minimal checking that it isn't a completely broken package during a PR that would update the image. Seems ok to  me

Reply:
Thanks @hop-dev . It seems ok with me as well, so I'll 👍 it.